conformance/lang/expressions/unary-expression/unary_complement_expression.balt

```diff
@@ -0,0 +1,523 @@
+Test-Case: output
+Description: Test unary complement basic operations.
+Labels: DecimalNumber, int, unary-complement
+
+function init() {
+    io:println(~0); // @output -1
+
+    int a1 = ~1;
+    io:println(a1); // @output -2
+
+    io:println(~~0 == 0); // @output true
+}
+
+Test-Case: output
+Description: Test unary complement with hex literal.
+Labels: HexIntLiteral, unary-complement
+
+function init() {
+    io:println(~0x0FFF); // @output -4096
+}
+
+Test-Case: output
+Description: Test unary complement operator's operand and result static types.
+Labels: DecimalNumber, function-call-expr, unary-complement
+
+function init() {
+    io:println(getComplement(10)); // @output -11
+}
+
+public function getComplement(int i) returns int {
+    return ~i;
+}
+
+Test-Case: output
+Description: Test unary complement's bit inversion.
+Labels: additive-expr, DecimalNumber, function-call-expr, unary-complement, unary-minus
+
+function init() {
+    io:println(~5 == getTwosComplementRepresentation(5)); // @output true
+    io:println(~-5 == getTwosComplementRepresentation(-5)); // @output true
+    io:println(~0 == getTwosComplementRepresentation(0)); // @output true
+}
+
+function getTwosComplementRepresentation(int x) returns int {
+    return -(x + 1);
+}
+
+Test-Case: output
+Description: Test unary complement result type.
+Labels: DecimalNumber, int, is-expr, unary-complement
+
+function init() {
+    int a1 = ~0;
+    io:println(a1); // @output -1
```

Comment:
Shouldn't we check `a1 is int` also?

Reply:
at compile-time, it's anyway restricted to int from RHS, isn't it? I have added a 'var' case to check result type 'is int'?